web/src/main/java/com/marklogic/hub/web/service/EntityManagerService.java

```diff
@@ -142,9 +143,12 @@ public EntityModel saveEntity(EntityModel entity) throws IOException {
     }
 
     public void deleteEntity(String entity) throws IOException {
-        File entitiesFile = hubConfig.getHubEntitiesDir().resolve(entity + ENTITY_FILE_EXTENSION).toFile();
+        String entityFileName = entity + ENTITY_FILE_EXTENSION;
```

Comment:
I think this is the right method to expose, and I wouldn't yet expose a general-purpose "deleteDocument" method until there's at least one more use case for it. 

The concept of deleting a document isn't really specific to a data hub, it's just a general ML feature. The concept of deleting an entity is specific to a data hub, and so this method makes a lot of sense and seems very handy. 

FWIW, I have no issue with this code deleting individual documents, I think that's the way to go.

Reply:
I made a general purpose "deleteDocument" to delete entity, flow, and step-def artifacts.
The current PR only deals with entity artifacts as per the scope of the bug but I think we should also delete the flow artifacts when they are deleted from the UI. (If agreed, I can take this up as a new task)
The step-defs can be shared among flows and thus we might only want to delete them if they are not backing any steps in any flows. (We can take this on when we introduce the concept of step-defs in the UI)